stable/pomerium/templates/authenticate-deployment.yaml

```diff
@@ -0,0 +1,141 @@
+{{- $secretName := default (include "pomerium.fullname" .) .Values.config.existingSecret }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ template "pomerium.name" . }}-{{ .Values.authenticate.name }}
+    chart: {{ template "pomerium.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "pomerium.fullname" . }}-{{ .Values.authenticate.name }}
```

Comment:
The name field in Kubernetes is limited to 63 characters. This is why the `fullname` function (see the helpers) truncates the name that way. By adding something to the end there is the possibility the name will be to long and cause install errors. We have encountered these which is why things are limited now. Can you please use functions for the various names and limit their length to avoid Kubernetes errors.

Reply:
Makes complete sense. Just a FYI, the docs do give what I had before as an example. Might be worth mentioning the character limit there as well.

>> The resource name should get the component as suffix (e. g. name: {{ include "myapp.fullname" . }}-server).